hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTxTest.java

```diff
@@ -0,0 +1,173 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.map.impl.nearcache;
+
+import com.hazelcast.cache.impl.nearcache.NearCache;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.proxy.NearCachedClientMapProxy;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.NearCacheConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.TransactionalMap;
+import com.hazelcast.map.nearcache.NearCacheTestSupport;
+import com.hazelcast.spi.properties.GroupProperty;
+import com.hazelcast.test.HazelcastParametersRunnerFactory;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.transaction.TransactionContext;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+import static java.lang.String.valueOf;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastParametersRunnerFactory.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ClientMapNearCacheTxTest extends NearCacheTestSupport {
+
+    @Parameterized.Parameter
+    public boolean batchInvalidationEnabled;
+
+    @Parameterized.Parameters(name = "batchInvalidationEnabled:{0}")
+    public static Iterable<Object[]> parameters() {
+        return Arrays.asList(new Object[]{Boolean.TRUE}, new Object[]{Boolean.FALSE});
+    }
+
+    protected final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.shutdownAll();
+    }
+
+    @Test
+    public void test_whenEmptyMap_thenPopulatedNearCacheShouldReturnNull_neverNULL_OBJECT() {
+        int size = 10;
+        String mapName = randomMapName();
+
+        NearCacheConfig nearCacheConfig = newNearCacheConfig();
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(nearCacheConfig);
+
+        HazelcastInstance client = getNearCachedClient(mapName, config, nearCacheConfig);
+
+        // populate map
+        TransactionContext transactionContext = client.newTransactionContext();
+        transactionContext.beginTransaction();
+        TransactionalMap<Integer, Integer> map = transactionContext.getMap(mapName);
+        for (int i = 0; i < size; i++) {
+            // populate Near Cache
+            assertNull(map.get(i));
+            // fetch value from Near Cache
+            assertNull(map.get(i));
+        }
+        transactionContext.commitTransaction();
+    }
+
+    @Test
+    public void testBasicUsage() {
```

Comment:
Personally I'm not crazy about such tests since you do, verify, do, verify, do.

Each of these steps deserves its own test.